src/Core/Domain/Product/Pack/QueryResult/PackedProductDetails.php

```diff
@@ -0,0 +1,145 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\Pack\QueryResult;
+
+/**
+ * Holds packed product data
+ */
+class PackedProductDetails
+{
+    /**
+     * @var int
+     */
+    private $productId;
+
+    /**
+     * @var string
+     */
+    private $productName;
+
+    /**
+     * @var int
+     */
+    private $quantity;
+
+    /**
+     * @var int
+     */
+    private $combinationId;
+
+    /**
+     * @var string
+     */
+    private $reference;
+
+    /**
+     * @var ?string
+     */
+    private $imageUrl = null;
+
+    /**
+     * @param int $productId
+     * @param int $quantity
+     * @param int $combinationId
+     * @param string $productName
+     * @param string $reference
+     * @param ?string $imageUrl
+     */
+    public function __construct(
+        int $productId,
+        int $quantity,
+        ?int $combinationId,
+        string $productName,
+        string $reference,
+        ?string $imageUrl
```

Comment:
```suggestion
        string $imageUrl
```

Reply:
These two parameters are not nullable Combination ID can have the 0 value, and imageUrl is always filled even if it's the fallback image